ntbs-service/DataAccess/NotificationRepository.cs

```diff
@@ -379,7 +379,7 @@ public async Task<IEnumerable<Notification>> GetInactiveNotificationsToCloseAsyn
                 .Include(n => n.TreatmentEvents)
                     .ThenInclude(t => t.TreatmentOutcome)
                 .Where(n => n.NotificationStatus == NotificationStatus.Notified)
-                .Where(n => n.TreatmentEvents.Any(t => t.TreatmentOutcome != null))
+                .Where(n => n.TreatmentEvents.Any(t => t.TreatmentOutcome != null && t.EventDate < DateTime.Today.AddYears(-1)))
```

Comment:
It could be worth adding a comment here that explains that this query will return all results where `ShouldBeClosed` is `true` as well as some others where it is `false`, but that we're doing this to avoid pulling as many notifications into memory as possible.